lightblue/src/main/java/org/esbtools/eventhandler/lightblue/LightblueNotificationRepository.java

```diff
@@ -42,27 +42,25 @@
 public class LightblueNotificationRepository implements NotificationRepository {
     private final LightblueClient lightblue;
     private final String[] entities;
-    private final Locking locking;
+    private final LockStrategy lockStrategy;
     private final Map<String, NotificationFactory> notificationFactoryByEntityName;
     private final Clock clock;
 
     public LightblueNotificationRepository(LightblueClient lightblue, String[] entities,
-            String lockingDomain, Map<String, NotificationFactory> notificationFactoryByEntityName,
-            Clock clock) {
+            LockStrategy lockStrategy,
+            Map<String, NotificationFactory> notificationFactoryByEntityName, Clock clock) {
         this.lightblue = lightblue;
         this.entities = entities;
         this.notificationFactoryByEntityName = notificationFactoryByEntityName;
         this.clock = clock;
-
-        locking = lightblue.getLocking(lockingDomain);
+        this.lockStrategy = lockStrategy;
     }
 
     @Override
     public List<LightblueNotification> retrieveOldestNotificationsUpTo(int maxEvents)
-            throws LightblueException {
-        try {
-            blockUntilLockAcquired(Locks.forNotificationsForEntities(entities));
-
+            throws Exception {
+        try (LockedResource lock = lockStrategy
+                .blockUntilAcquired(ResourceIds.forNotificationsForEntities(entities))) {
```

Comment:
We had talked about that initially, but I ran into some issues, at least for document event entities. I think we could make all of the synchronization more robust, but it's a question of what is appropriate for 1.0.0, given the time pressure? I'm more concerned with it working and reducing chance of lost / stuck events than performance right now. A matter of prioritization.